indico/modules/events/contributions/controllers/management.py

```diff
@@ -459,3 +466,108 @@ def _process(self):
         self.event_new.log(EventLogRealm.management, EventLogKind.negative, 'Contributions',
                            'Deleted type: {}'.format(self.contrib_type.name), session.user)
         return jsonify_data(flash=False)
+
+
+class RHManageContributionFields(RHManageContributionsBase):
+    """Dialog to manage the custom contribution fields of an event"""
+
+    def _process(self):
+        custom_fields = self.event_new.contribution_fields.order_by(ContributionField.position).all()
+        custom_field_types = sorted(get_contrib_field_types().values(), key=attrgetter('friendly_name'))
+        return jsonify_template('events/contributions/management/fields_dialog.html', event=self.event_new,
+                                custom_fields=custom_fields, custom_field_types=custom_field_types)
+
+
+class RHSortContributionFields(RHManageContributionsBase):
+    """Sort the custom contribution fields of an event"""
+
+    def _process(self):
+        field_by_id = {field.id: field for field in self.event_new.contribution_fields.all()}
+        field_ids = map(int, request.form.getlist('field_ids'))
+        for index, field_id in enumerate(field_ids, 0):
+            field_by_id[field_id].position = index
+            del field_by_id[field_id]
+        for index, field in enumerate(sorted(field_by_id.values(), key=attrgetter('position')), len(field_ids)):
```

Comment:
the only case where that would iterate over anything is when someone adds a new field at the same time. seems like a bit of overkill to handle that edge case. anyway, since it's there now i wouldn't remove it